### DISCUSSION TITLE
Reference is Undefined if Used inside Template

### DISCUSSION THREAD

**Discussion by alexanderbluhm** [alpinejs/alpine#2494] [7. Bugs (archived)]
In my application I have an image gallery. If I click on an image a modal opens showing the image and a slider to go to the prev and next image. However, I want the clicked image to be shown in the slider. So, I want to call `show(imageId)` on click to move the slider to the correct position. The simplified html structure looks like this:

```html
<div x-data="
    show() {
      this.$refs.slider.scrollTo(...)
    }
  ">
  <!-- Gallery -->
  <div>
    <button @click="show(2)">...</button>
  </div>

  <!-- Modal -->
  <template x-teleport="body">
    <!-- Image Slider here -->
    <ul x-ref="slider" tabindex="0">...</ul>
  </template>
</div>
```

The problem now is that I can't access `this.$refs.slider` because of the `x-teleport="body"`.
If I remove the template it works as expected. Here is a minimal reproduction: https://codepen.io/il4miy/pen/QWqdxqm?editors=1001

Would be awesome if usage of refs through `x-teleport` would be possible.

(I temporarily solved my issue by deleting the `x-teleport`).

**Comment by stepanjakl** [comment]
I was facing a similar issue, where the `x-data` inside the template wouldn't be available (and throw errors). I used this plugin instead, and it's working as expected: https://github.com/motekar/alpinejs-super-teleport.

**Reply by SimoTod** [reply]
It's because you are not supposed to teleport the template (which moves it outside the component scope) but you should teleport the element inside the template instead

**Reply by ekwoka** [reply]
That plugin actually doesn't do anything that x-teleport doesn't. Well, maybe the replace. But the rest are in x-teleport
